airbyte-integrations/connectors/source-shopify/source_shopify/streams/base_streams.py

```diff
@@ -45,6 +45,7 @@ def __init__(self, config: Dict) -> None:
         super().__init__(authenticator=config["authenticator"])
         self._transformer = DataTypeEnforcer(self.get_json_schema())
         self.config = config
+        self._current_limit = self.limit  # Dynamic limit initialized to default
```

Comment:
Assuming that we have a 1 to 1 relation between a ShopifyStream and an error handler, why does the `current_limit` need to live on the `stream` instead of the `error_handler`? The part that stuck out to me reviewing the code is that we need to store the entire stream within the LimitReducingErrorHandler class. I think we should ideally try to find a way to avoid doing that unless absolutely necessary. 

If the main piece of functionality we want is to track a dynamically changing limit, we should try to store that where we apply it. I'll add a few sub-comments, in other parts of the code. Ultimately if we have to store it on the stream and pass it to the handler, its not the end of the world, but I want to at least see if we can brainstorm a better way